src/compiler/checker.ts

```diff
@@ -6052,6 +6066,28 @@ namespace ts {
                 }
                 return Ternary.True;
             }
+
+            function constructorRelatedTo(sourceSignature: Signature, targetSignature: Signature, reportErrors: boolean) {
+                if (sourceSignature && targetSignature && sourceSignature.declaration && targetSignature.declaration) {
+                    // A public, protected and private signature is assignable to a private signature.
+                    // A public and protected signature is assignable to a protected signature.
+                    // And only a public signature is assignable to public signature.
+                    const sourceAccessibility = sourceSignature.declaration.flags & (NodeFlags.Private | NodeFlags.Protected);
+                    const targetAccessibility = targetSignature.declaration.flags & (NodeFlags.Private | NodeFlags.Protected);
+
+                    const isRelated = targetAccessibility === NodeFlags.Private
+                    || (targetAccessibility === NodeFlags.Protected && sourceAccessibility !== NodeFlags.Private)
+                    || (targetAccessibility !== NodeFlags.Protected && !(sourceAccessibility & (NodeFlags.Private | NodeFlags.Protected)));

```

Comment:
Why `sourceAccessibility & (NodeFlags.Private | NodeFlags.Protected)`? Isn't that just `sourceAccessibility` as it is?